src/GafferScene/MotionPath.cpp

```diff
@@ -0,0 +1,380 @@
+{
+	if( path.empty() || !(filterValue( context ) & PathMatcher::ExactMatch) )
+	{
+		return inPlug()->objectPlug()->defaultValue();
+	}
+
+	float start = ( (FrameMode)startModePlug()->getValue() == FrameMode::Absolute ) ? startFramePlug()->getValue() : context->getFrame() + startFramePlug()->getValue();
+	float end = ( (FrameMode)endModePlug()->getValue() == FrameMode::Absolute ) ? endFramePlug()->getValue() : context->getFrame() + endFramePlug()->getValue();
+	if( start >= end )
+	{
+		return inPlug()->objectPlug()->defaultValue();
+	}
+
+	float increment = incrementPlug()->getValue();
+	int steps = ( end - start ) / increment + 1;
+
+	V3fVectorDataPtr points = new V3fVectorData;
+	points->setInterpretation( GeometricData::Point );
+	auto &p = points->writable();
+	p.reserve( steps );
+
+	QuatfVectorDataPtr orientations = new QuatfVectorData;
+	auto &orients = orientations->writable();
+	orients.reserve( steps );
+
+	FloatVectorDataPtr sampleFrames = new FloatVectorData;
+	auto &frames = sampleFrames->writable();
+	frames.reserve( steps );
+
+	Imath::M44f transform;
+	Context::EditableScope scope( context );
+	for( float f = start; f < end; f += increment )
+	{
+		scope.setFrame( f );
+		transform = inPlug()->fullTransform( path );
+		p.emplace_back( transform.translation() );
+		orients.emplace_back( Imath::extractQuat( transform ) );
+		frames.emplace_back( f );
+	}
+
+	scope.setFrame( end );
+	transform = inPlug()->fullTransform( path );
+	p.emplace_back( transform.translation() );
+	orients.emplace_back( Imath::extractQuat( transform ) );
+	frames.emplace_back( end );
+
+	CurvesPrimitivePtr motionPath = new CurvesPrimitive( new IntVectorData( { (int)p.size() } ), CubicBasisf::linear(), false, points );
+	motionPath->variables["orientation"] = PrimitiveVariable( PrimitiveVariable::Vertex, orientations );
+	motionPath->variables["frame"] = PrimitiveVariable( PrimitiveVariable::Vertex, sampleFrames );
+	return motionPath;
+}
+
+void MotionPath::hashSetNames( const Context *context, const ScenePlug *parent, MurmurHash &h ) const
+{
+	FilteredSceneProcessor::hashSetNames( context, parent, h );
+
+	isolatedInPlug()->setNamesPlug()->hash( h );
+}
+
+ConstInternedStringVectorDataPtr MotionPath::computeSetNames( const Context *context, const ScenePlug *parent ) const
+{
+	ConstInternedStringVectorDataPtr setNames = isolatedInPlug()->setNamesPlug()->getValue();
+
+	InternedStringVectorDataPtr result = new InternedStringVectorData;
+	for( auto &setName : setNames->readable() )
+	{
+		if( setName == g_camerasSetName || setName == g_lightsSetName || setName == g_defaultLightsSetName )
+		{
+			continue;
+		}
+
+		result->writable().emplace_back( setName );
+	}
+
+	return result;
+}
+
+void MotionPath::hashSet( const InternedString &setName, const Context *context, const ScenePlug *parent, MurmurHash &h ) const
+{
+	FilteredSceneProcessor::hashSet( setName, context, parent, h );
+
+	h.append( setName );
+
+	isolatedInPlug()->setPlug()->hash( h );
```

Comment:
Addressed in caac9ef8fe7201d5b791c4742874fa05ebca6996

Reply:
Since the compute is a direct pass-through after this check has been performend, we can also do `h = isolatedInPlug()->setPlug()->hash()` for the non camera/light case.

Reply:
Addressed in 3d09e67